charms/kubernetes_snaps.py

```diff
@@ -544,5 +577,6 @@ def write_etcd_client_credentials(ca, cert, key):
 
 def write_service_account_key(key):
     dest = "/root/cdk/serviceaccount.key"
+    os.makedirs("/root/cdk", exist_ok=True)
     with open(dest, "w") as f:
         f.write(key)
```

Comment:
who doesn't love `Path`

```suggestion
def write_service_account_key(key:str) -> None
    dest = Path("/root/cdk/serviceaccount.key")
    dest.parent.mkdir(mode=0o755, exist_ok=True)
    dest.write_text(key)
```